homeassistant/components/aurora_abb_powerone/__init__.py

```diff
@@ -1 +1,73 @@
 """The Aurora ABB Powerone PV inverter sensor integration."""
+
+# Reference info:
+# https://s1.solacity.com/docs/PVI-3.0-3.6-4.2-OUTD-US%20Manual.pdf
+# http://www.drhack.it/images/PDF/AuroraCommunicationProtocol_4_2.pdf
+
+import asyncio
+from collections import defaultdict
+import logging
+
+from aurorapy.client import AuroraSerialClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ADDRESS, CONF_PORT
+from homeassistant.core import HomeAssistant
+
+from .const import ATTR_SERIAL_NUMBER, DOMAIN
+
+PLATFORMS = ["sensor"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Aurora ABB PowerOne component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Aurora ABB PowerOne from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+    _LOGGER.debug("async_setup_entry got user config data=%s", entry.data)
+
+    comport = entry.data[CONF_PORT]
+    address = entry.data[CONF_ADDRESS]
+    client = AuroraSerialClient(address, comport, parity="N", timeout=1)
+    all_device_params = [
+        {"type": "sensor", "parameter": "instantaneouspower", "name": "Power Output"},
+        {"type": "sensor", "parameter": "temperature", "name": "Temperature"},
+    ]
+
+    entry_data = hass.data[DOMAIN][entry.entry_id] = {
+        "client": client,
+        "devices": defaultdict(list),
+    }
+    serialnum = entry.data[ATTR_SERIAL_NUMBER]
+    entry_data["devices"][serialnum] = defaultdict(list)
```

Comment:
`entry_data["devices"]` defaults each value to a list but here you're overriding that to be a dict. I also don't see you using it as a default dict anywhere. Why not just instantiate it with `{}` here and above?

Reply:
Fixed.